Neuro/Vision/VisionHandler.cs

```diff
@@ -1,39 +1,60 @@
-﻿using Neuro.Utils;
-using System;
+﻿using System;
 using System.Collections.Generic;
+using System.Linq;
+using Neuro.DependencyInjection;
+using Neuro.Vision.DataStructures;
+using Reactor.Utilities.Attributes;
 using UnityEngine;
 
-namespace Neuro;
+namespace Neuro.Vision;
 
-public class Vision
+// TODO: Refactor this entire class
+[RegisterInIl2Cpp]
+public class VisionHandler : MonoBehaviour, IVisionHandler
 {
-    public List<DeadBody> deadBodies = new List<DeadBody>();
+    public VisionHandler(IntPtr ptr) : base(ptr) { }
 
-    public Dictionary<PlayerControl, LastSeenPlayer> playerLocations = new Dictionary<PlayerControl, LastSeenPlayer>();
+    public IContextProvider Context { get; set; }
 
-    public Dictionary<byte, PlayerControl> playerControls = new Dictionary<byte, PlayerControl>();
+    public Vector2 DirectionToNearestBody { get; set; }
 
-    public float roundStartTime = 0f; // in seconds
-    public float lastPlayerUpdateTime = 0f; // in seconds, records the last time PlayerControl.FixedUpdate was called
-    public float lastPlayerUpdateDuration = 0f; // in seconds, records the time elapsed since last PlayerControl.FixedUpdate was called
+    private readonly List<DeadBody> deadBodies = new();
+    private readonly Dictionary<byte, PlayerControl> playerControls = new(); // TODO: Use GameData.GetPlayerById
+    private readonly Dictionary<PlayerControl, LastSeenPlayer> playerLocations = new(); // TODO: Don't use PlayerControls as dictionary keys,
```

Comment:
I believe we could use `PlayerControl.NetId` (uint) for the dictionary key. Another option could be `PlayerControl.Puid` (string) but all dummies have null/empty Puids so that would make offline testing much harder.

Reply:
`PlayerControl.PlayerId` should be just fine

Reply:
Ah, I missed that. That would probably be best.